iset.mm

```diff
@@ -65455,6 +65455,104 @@ elements or fails to hold (is equal to ` (/) ` ) for some element.
       cvv dom2 ) BCAADZEFBGZHZCGZHZQTAIZSEUATAJTAKLUDUAUCMTUBMNUBAITUBAOPR $.
   $}
 
+  ${
+    $d A x $.  $d B x $.  $d ph x $.
+    exmidfodomrlemeldju.a $e |- ( ph -> A C_ 1o ) $.
+    exmidfodomrlemeldju.el $e |- ( ph -> B e. ( A |_| 1o ) ) $.
+    $( Lemma for ~ exmidfodomr .  A variant of ~ djur .  (Contributed by Jim
+       Kingdon, 2-Jul-2022.) $)
+    exmidfodomrlemeldju $p |- ( ph
+        -> ( B = ( inl ` (/) ) \/ B = ( inr ` (/) ) ) ) $=
```

Comment:
Again, I would write
```
-> ( B = ( ( inl |` A ) ` (/) ) \/ B = ( ( inr |` 1o ) ` (/) ) )
```
since I think the needed theorems with the restrictions are all available, and a consistent use of the restricted injections is possible. But this is only a lemma, and maybe the proofs of the lemmas up to the main theorem would be more cumbersome (even though clearer, to me) ?

Reply:
Already on this branch as `exmidfodomrlemeldjuN` (excerpt below).

I had in mind keeping both versions around (because I agree with "cumbersome" on the one side but also want to find a good way to deal with the other being "clearer" for those who find that). However, I'm not sure I had great ideas about naming. The `N` convention was intended to be temporary. For the case of `exmidfodomrlemrN` it could be an `ALT` because it proves the same theorem as `exmidfodomrlemr` but with a different proof. However, as far as I know we don't use the `ALT` convention where the theorem being proved is different. So maybe we include `res` in the names of the versions with the restrictions? I mean, sometimes in this sort of situation we'd put one of the choices in a mathbox but to do that we'd need to decide which one goes in main.

```
  ${
    $d A x $.  $d B x $.  $d ph x $.
    exmidfodomrlemeldjuN.a $e |- ( ph -> A C_ 1o ) $.
    exmidfodomrlemeldjuN.el $e |- ( ph -> B e. ( A |_| 1o ) ) $.
    $( Lemma for ~ exmidfodomr .  A variant of ~ eldju .  (Contributed by Jim
       Kingdon, 9-Jul-2022.) $)
    exmidfodomrlemeldjuN $p |- ( ph
        -> ( ( (/) e. A /\ B = ( ( inl |` A ) ` (/) ) )
          \/ B = ( ( inr |` 1o ) ` (/) ) ) ) $=
```

Reply:
Ah, missed it ! Great minds think alike ! (Now I understand why the cumbersome `(/) e. A` is needed, simply because `(/)` was chosen to represent an undefined value.)

Don't worry about the label.  I chose to suffix an "r" for "restriction", for lack of a better option, but probably we won't keep both in Main, and we can change that later.

Reply:
> Now I understand why the cumbersome `(/) e. A` is needed, simply because `(/)` was chosen to represent an undefined value.

I don't think it has anything to do with undefined values here. If we were doing the proof with `{ 0 , 1 }` rather than `{ (/) , 1o }` it would be `0` not `(/)`. I described this at https://github.com/metamath/set.mm/pull/2669#discussion_r917289132 and it is why avoiding `djulclb` in the `exmidfodomrlemrN` proof is not just a matter of slotting in a slightly different variant of `djulclb` - you need to carry around `(/) e. A` to implement the suggestion of "this is an argument for proving C e. A first" (in the words of https://github.com/metamath/set.mm/pull/2669#discussion_r917302632 )

Reply:
Ah yes, you're right, I remember.